presto-main/src/test/java/com/facebook/presto/type/TestDoubleOperators.java

```diff
@@ -177,6 +178,20 @@ public void testCastToBigint()
     {
         assertFunction("cast(37.7E0 as bigint)", BIGINT, 38L);
         assertFunction("cast(17.1E0 as bigint)", BIGINT, 17L);
+
+        assertFunction("cast(" + Math.nextDown(0x1.0p63) + " as bigint)", BIGINT, (long) Math.nextDown(0x1.0p63));
+        assertFunction("cast(" + Math.nextUp(-0x1.0p63) + " as bigint)", BIGINT, (long) Math.nextUp(-0x1.0p63));
+
+        assertFunction("cast(9.2E18 as bigint)", BIGINT, 9200000000000000000L);
+        assertFunction("cast(-9.2E18 as bigint)", BIGINT, -9200000000000000000L);
+        assertFunction("cast(2.21E9 as bigint)", BIGINT, 2210000000L);
+        assertFunction("cast(-2.21E9 as bigint)", BIGINT, -2210000000L);
+
+        assertInvalidFunction("cast(9.3E18 as bigint)", NUMERIC_VALUE_OUT_OF_RANGE);
+        assertInvalidFunction("cast(-9.3E18 as bigint)", NUMERIC_VALUE_OUT_OF_RANGE);
+
+        assertInvalidFunction("cast(" + 0x1.0p63 + " as bigint)", NUMERIC_VALUE_OUT_OF_RANGE);
+        assertInvalidFunction("cast(" + -0x1.0p63 + " as bigint)", NUMERIC_VALUE_OUT_OF_RANGE);
```

Comment:
This should succeed. If it doesn't, it means your implementation is not correct.